Reorder docs and add a landing page (#66)
PrefectHQ/prefect-dask: README.md

        <img src="https://img.shields.io/badge/discourse-browse_forum-red.svg?color=0052FF&labelColor=090422&logo=discourse" /></a>
</p>

![DALL·E 2023-01-10 15 19 31](https://user-images.githubusercontent.com/15331990/211682578-3e341709-6509-4c95-a6af-3b1160fe2961.png)

Visit the full docs [here](https://PrefectHQ.github.io/prefect-dask) to see additional examples and the API reference.

The `prefect-dask` collection makes it easy to include distributed processing for your flows. Check out the examples below to get started!

## Integrate with Prefect flows

> Perhaps you're already working with Prefect flows. It downloads many images to train your machine learning model--the problem is that it runs slow because it's executed sequentially.

Reviewer: ```suggestion
Perhaps you're already working with Prefect flows. Say your flow downloads many images to train your machine learning model. Unfortunately, the it takes a long time to download your flows because your code is running sequentially.
```

> With `prefect-dask`, you can parallelize your flow, and the best part is that it's as easy as:

Reviewer: ```suggestion
After installing `prefect-dask` you can parallelize your flow in three simple steps:
```

> 1. Adding the import: `from prefect_dask import DaskTaskRunner`

Reviewer: ```suggestion
1. Add the import: `from prefect_dask import DaskTaskRunner`
```

2. Specifying the task runner in the flow decorator: `@flow(task_runner=DaskTaskRunner)`
3. Submitting tasks to the flow's task runner: `a_task.submit(*args, **kwargs)`

> 

Reviewer: ```suggestion
The parallelized code  runs in about 1/3 of the time in our test!  And that's without distributing the workload over multiple machines.
```

Here's the before and after!

=== "Before"

    ```python hl_lines="1"
    # Completed in 15.2 seconds

> 

Reviewer: Maybe adding an collapse/expand box for the code samples would be good. Probably not.

Author: I think it's better visible by default

    URL_FORMAT = (
        "https://www.cpc.ncep.noaa.gov/products/NMME/archive/"
        "{year:04d}{month:02d}0800/current/images/nino34.rescaling.ENSMEAN.png"
    )

    @task
    def download_image(year: int, month: int, directory: Path) -> Path:
        # download image from URL
        url = URL_FORMAT.format(year=year, month=month)
        resp = httpx.get(url)

        # save content to directory/YYYYMM.png
        file_path = (directory / url.split("/")[-1]).with_stem(f"{year:04d}{month:02d}")
        file_path.write_bytes(resp.content)
        return file_path

    @flow(task_runner=DaskTaskRunner(cluster_kwargs={"processes": False}))
    def download_nino_34_plumes_from_year(year: int) -> List[Path]:
        # create a directory to hold images
        directory = Path("data")
        directory.mkdir(exist_ok=True)

        # download all images
        file_paths = []
        for month in range(1, 12 + 1):
            file_path = download_image.submit(year, month, directory)
            file_paths.append(file_path)
        return file_paths

    if __name__ == "__main__":
        download_nino_34_plumes_from_year(2022)
    ```

> The original flow completes in **15.2 seconds**, which is okay.

Reviewer: ```suggestion
The original flow completes in 15.2 seconds.
```

However, with just a few minor tweaks, we were able to reduce the runtime by nearly **three** folds, down to just **5.7** seconds!

## Integrate with Dask client/cluster and collections

Suppose you have an existing Dask client/cluster and collection, like a `dask.dataframe.DataFrame`, and you want to add observability.

With `prefect-dask`, there's no major overhaul necessary because Prefect was designed with incremental adoption in mind! It's as easy as:

> 

Author: I think I mentioned it: "you want to add observability."?

        <img src="https://img.shields.io/badge/discourse-browse_forum-red.svg?color=0052FF&labelColor=090422&logo=discourse" /></a>
</p>

Visit the full docs [here](https://PrefectHQ.github.io/prefect-dask) to see additional examples and the API reference.

The `prefect-dask` collection makes it easy to include distributed processing for your flows. Check out the examples below to get started!

## Integrate with Prefect flows

> Perhaps you're already working with Prefect flows. Say your flow downloads many images to train your machine learning model. Unfortunately, the it takes a long time to download your flows because your code is running sequentially.

Author: ```suggestion
Perhaps you're already working with Prefect flows. Say your flow downloads many images to train your machine learning model. Unfortunately, it takes a long time to download your flows because your code is running sequentially.
```
